content/learning-paths/servers-and-cloud-computing/glibc-with-lse/mongo_start.md

```diff
@@ -0,0 +1,95 @@
+---
+# User change
+title: "Start MongoDB utilizing the newly built Glibc with LSE"
+weight: 3 # 1 is first, 2 is second, etc.
+
+# Do not modify these elements
+layout: "learningpathall"
+---
+
+You are now ready to start MongoDB utilizing the newly built glibc with LSE on your Arm server.
+
+
+## MongoDB Setups
+Build and install MongoDB version `5.3.2`:
+```console
+#for Fedora/RHEL
+sudo yum install -y git gcc g++ python3-devel openssl-devel libcurl-devel 
+#for Ubuntu/Debian
+sudo apt install -y git gcc g++ python-dev-is-python3 python3-pip libssl-dev libcurl4-openssl-dev liblzma-dev
+
+export PYTHON_KEYRING_BACKEND=keyring.backends.null.Keyring
+
+cd ~
+sudo git clone https://github.com/mongodb/mongo.git 
+cd mongo
+sudo git checkout r5.3.2
+sudo python3 -m pip install -r etc/pip/compile-requirements.txt
+sudo python3 buildscripts/scons.py install-mongod -j$(expr $(nproc) - 1) --disable-warnings-as-errors
+```
+
+## MongoDB Starts utilizing the newly built Glibc with LSE
+```console
+cd ~
+mkdir -p ~/mongodb-5.3.2/data
+vi ~/mongodb-5.3.2/mongodb.conf
+
+dbpath=mongodb-5.3.2/data
+logpath=mongodb-5.3.2/mongodb.log
+pidfilepath=mongodb-5.3.2/data/mongo.pid
+fork=true
```

Comment:
With fork=true in the mongodb.conf file I run into the following error:

~/glibc-2.32_build_install/build/testrun.sh ~/mongo/build/install/bin/mongod -f ~/mongodb-5.3.2/mongodb.conf --wiredTigerCacheSizeGB=20
about to fork child process, waiting until server is ready for connections.
forked process: 76714
ERROR: child process failed, exited with 1
To see additional information in this output, start without the "--fork" option.

It runs fine when I remove the fork option

Reply:
Both "fork=true" and "fork=false" are OK.

But I think the following information to be different from what you see:

cd ~
mkdir -p /mongodb-5.3.2/data
vi /mongodb-5.3.2/mongodb.conf

dbpath=/mongodb-5.3.2/data
logpath=/mongodb-5.3.2/mongodb.log
pidfilepath=/mongodb-5.3.2/data/mongo.pid
fork=true
logappend=true
bind_ip=0.0.0.0
port=27017